bundle-workflow/src/ci_workflow/check.py

```diff
@@ -8,17 +8,10 @@
 
 
 class Check(ABC):
-    def __init__(self, component, git_repo, version, arch, snapshot):
+    def __init__(self, component, git_repo, target):
         self.component = component
         self.git_repo = git_repo
-        self.version = version
-        self.arch = arch
-        self.snapshot = snapshot
-        self.opensearch_version = version + "-SNAPSHOT" if snapshot else version
-        self.component_version = version + ".0-SNAPSHOT" if snapshot else f"{version}.0"
-        if self.component.name == "OpenSearch":
-            # HACK: OpenSearch version is 3-digits
```

Comment:
Would this part be needed in the `BuildTarget` implementation?

Reply:
It was ported into the check itself where we treat OpenSearch itself as a component. The build target defines the version that a component _should_ be.

Reply:
Here: https://github.com/opensearch-project/opensearch-build/pull/443/files/4aa773ceaeea391b9451ec182c128f9a8740e5aa#diff-a8e1cc3a88bb0cf63acc8cec63acc8c5d1b232896f05f3ad02c12a4783613d31R56

Reply:
Got it! Thanks!